tests/unit/io/test_libevreactor.py

```diff
@@ -296,3 +295,37 @@ def test_partial_message_read(self, *args):
 
         self.assertTrue(c.connected_event.is_set())
         self.assertFalse(c.is_defunct)
+
+    def test_watchers_are_finished(self, *args):
```

Comment:
This test has been moved to unit tests so we can patch more things, it was messing up with the environment in the integration tests.